### PR TITLE
Feature/bruteforce to mr estimator

### PR DESCRIPTION
**Pull request by Javierverbel**
### Description
- Bruteforce algorithm for MinRank Estimaor

### CODE REVIEW

File: cryptographic_estimators/MREstimator/MRAlgorithms/bruteforce.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+from ...MREstimator.mr_problem import MRProblem
+from ...base_algorithm import optimal_parameter
+from math import log2, ceil
+from math import comb as binomial
+from ..mr_helper import minors_polynomial
+from ..mr_constants import MR_NUMBER_OF_KERNEL_VECTORS_TO_GUESS, MR_NUMBER_OF_COEFFICIENTS_TO_GUESS
+
+
+class BruteForce(MRAlgorithm):
+    r"""
```
Comment (FloydZ) [inline cryptographic_estimators/MREstimator/MRAlgorithms/bruteforce.py:28]: why interpret it as a raw string? I think the `r` should be removed

File: cryptographic_estimators/MREstimator/MRAlgorithms/bruteforce.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+        sage: from cryptographic_estimators.MREstimator.MRAlgorithms.bruteforce import BruteForce
+        sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+        sage: E = BruteForce(MRProblem(q=7, m=9, n=10, k=15, r=4))
+        sage: E
+        BruteForce estimator for the MinRank problem with (q, m, n, k, r) = (7, 9, 10, 15, 4)
+    """
+
+    def __init__(self, problem: MRProblem, **kwargs):
+
```
Comment (FloydZ) [inline cryptographic_estimators/MREstimator/MRAlgorithms/bruteforce.py:48]: a newline to much

File: cryptographic_estimators/MREstimator/MRAlgorithms/bruteforce.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.bruteforce import BruteForce
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: BFE = BruteForce(MRProblem(q=16, m=15, n=15, k=78, r=6))
+            sage: BFE.lv()
+            0
+        """
+        return self._get_optimal_parameter(MR_NUMBER_OF_COEFFICIENTS_TO_GUESS)
+
+    def _BFE_time_complexity_helper_(self, q, k_reduced, r):
+        time = 0
```
Comment (FloydZ) [inline cryptographic_estimators/MREstimator/MRAlgorithms/bruteforce.py:103]: Although thats an internal function: maybe its beneficial if it had a doc string and types at the arguments.

File: cryptographic_estimators/MREstimator/MRAlgorithms/minors.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+        sage: from cryptographic_estimators.MREstimator.MRAlgorithms.minors import Minors
+        sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+        sage: E = Minors(MRProblem(q=7, m=9, n=10, k=15, r=4))
+        sage: E
+        Minors estimator for the MinRank problem with (q, m, n, k, r) = (7, 9, 10, 15, 4)
+    """
+
+    def __init__(self, problem: MRProblem, **kwargs):
+
```
Comment (FloydZ) [inline cryptographic_estimators/MREstimator/MRAlgorithms/minors.py:48]: newline

**Review by ricardovp85 — APPROVED** [review]
Apart from the minor changes requested by the other reviewer, I think no additional changes are needed.  It looks good to me.

File: cryptographic_estimators/MREstimator/MRAlgorithms/bruteforce.py (hunk 3 above)
Comment (Memphisd) [reply cryptographic_estimators/MREstimator/MRAlgorithms/bruteforce.py:106]: I agree, either it should have a descriptive name or a docstring

**Review by Memphisd — APPROVED** [review]
apart from minor comments mentioned by @FloydZ I think it is ready to be merged.

**Pull request merged**